control-plane/p4RuntimeSerializer.cpp

```diff
@@ -657,6 +658,15 @@ getMatchFields(const IR::P4Table* table, ReferenceMap* refMap, TypeMap* typeMap)
     return matchFields;
 }
 
+/// Parses P4Runtime-specific annotations.
+class ParseAnnotations : public P4::ParseAnnotations {
+ public:
+    ParseAnnotations() : P4::ParseAnnotations("P4Runtime") { }
+    virtual void postorder(IR::Annotation* annotation) override {
+        PARSE("controller_header", StringLiteral)
```

Comment:
This is just checking that the `@controller_header` "payload" is a string and transforming it to an IR StringLiteral, correct? The code would work the same without it?

If this sanity checks the annotation payload, can we remove this code later on:
```
auto nameConstant = controllerAnnotation->expr[0]->to<IR::StringLiteral>();
        if (nameConstant == nullptr) {
            ::error("@controller_header should be a string for declaration %1%", type);
            return;
        }
```

Also, why not do the other annotations, such as `@id`?

Reply:
Done. Added support for `@id` and `@hidden`. Removed the sanity check above and one for `@id`.